src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -153,6 +162,10 @@ public void Shutdown()
                         // Pass the exception to the task completion source, so the caller of the ExecuteAsync method can react
                         completion.SetException(exception);
 
+                        // Update the failure count for the request.
+                        // Failure details are captured separately in a non fatal watson report.
+                        _requestTelemetryLogger.LogFailure(methodName);
```

Comment:
Is it worth logging cancellation? In particular, in the case where the token is cancelled  (line 144) it might be useful to know how many requests are lost when the server shuts down, as another health indicator.

Reply:
I think a lot of requests can get cancelled, especially for things like completion.  But it probably can't hurt to add!